Lucia/update nerdpack detail docs

brammerl opened the pull request:
## Description

Updates guide images of the new nerdpack detail UI for the following articles:

- [Add your catalog metadata](https://developer.newrelic.com/build-apps/publish-deploy/catalog/)

- [Describe your app for the catalog](https://developer.newrelic.com/build-apps/ab-test/catalog/)

- [Create Hello World App](https://developer.newrelic.com/build-apps/build-hello-world-app/)

📢 NOTE 📢 
Please **do not merge** before we turn on the `DevEx/nerdpackItem_previews` ([link](https://staging-featureflag.datanerd.us/flags/8580/edit)) feature flag. That flag controls the UI displaying the images in the "What's inside" tab and we plan to enable it around the end of October once this ticket is complete: [[3rd party Nerdpacks] Add artifact level screenshots ](https://newrelic.atlassian.net/browse/DEVEX-2018)

## Reviewer Notes

If there are links or steps needed to test this work, add them here.

## Related Issue(s) / Ticket(s)

* [Ticket](https://newrelic.atlassian.net/secure/RapidBoard.jspa?rapidView=1581&projectKey=DEVEX&modal=detail&selectedIssue=DEVEX-2037)

## Screenshot(s)

If relevant, add screenshots here.

## Use Conventional Commits

Please help the maintainers by leveraging the following [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)
standards in your pull request title and commit messages.

## Use `chore`

* for minor changes / additions / corrections to content.
* for minor changes / additions / corrections to images.
* for minor non-functional changes / additions to github actions, github templates, package or config updates, etc

```bash
git commit -m "chore: adjusting config and content"
```

## Use `fix`

* for minor functional corrections to code.

```bash
git commit -m "fix: typo and prop error in the code of conduct"
```

## Use `feat`

* for major functional changes or additions to code.

```bash
git commit -m "feat(media): creating a video landing page"
```

tyreer commented on src/markdown-pages/build-apps/serve-publish-subscribe/catalog.mdx:
```diff
@@ -141,6 +141,10 @@ ls nerdlets/home/catalog
 
+After adding a screenshot into each directory, click your nerdpack to see the changes.
+
+![What's Inside Tab](../../../images/build-an-app/whats-inside-nerdlet-tab.png)
```
Do you think this will be a bit confusing to jump to the Account Maturity app when the other screenshots in this guide are referencing the Demo App?

I think we'll need to recreate this process for that Demo App for the sake of consistency

tyreer commented on src/markdown-pages/build-apps/serve-publish-subscribe/catalog.mdx:
````diff
@@ -141,6 +141,10 @@ ls nerdlets/home/catalog
 [output] {blue}screenshots
 ```
 
+After adding a screenshot into each directory, click your nerdpack to see the changes.
````
I think this text is a bit too early in these steps.

Looks like we've already got a step that covers adding the screenshots to the Nerdpack item directories. 

> STEP 7 OF 9
> Include screenshots in your root screenshots directory or any Nerdpack item screenshots directory.

And step 8, running `nr1 catalog:submit`,  would be required before being able to see these images in the UI.
 
How about in step 9, we add a new line and screenshot after this section?
> Click your Nerdpack to see the new details:

Maybe something like "Click the _What's inside_ tab to see any screenshots  you uploaded for individual items in your Nerdpack"

tyreer replied:
Also, took me a while to learn this one, but Nerdpack and Nerdlet are always capitalized. 😄

Here's a style guide reference that's been shared with us before:
https://docs.google.com/document/d/1cPV1hcNa5QQgyPNPJbIVL-1NwtP5fqoJubcs33xHKeI/edit#

tyreer commented:
Thanks for getting this going! 🙌

I think the work in `build-apps/serve-publish-subscribe/catalog.mdx` is exactly the kind of changes we're looking to introduce. But I wonder if there might be a few other guides that would benefit from updates as part of this work.

Were you able to check the 4 guides we updated in [this PR](https://github.com/newrelic/developer-website/pull/1597/) to see if any would benefit from any additional text that would highlight how Nerdpack item uploads are rendered to the What's inside tab?

I know @MichelLosier also collected a bunch of guides here that are worth double checking as well: https://newrelic.atlassian.net/browse/DEVEX-2036?focusedCommentId=974573

alexronquillo commented on src/markdown-pages/build-apps/create-an-ab-test-application/catalog-info.mdx:
```diff
@@ -106,6 +106,8 @@ Save the [this screenshot](https://github.com/newrelic-experimental/nru-programm
 
 Users will be able to see a carousel of screenshots in your app's **About** page in the catalog.
 
+![Screenshot Carousel](../../../images/ab-test/ab-testing-carousel-preview.png)
```
Could you elaborate on the two screenshot additions in this guide? The original intention here was not to have images because the user doesn't submit the catalog information until the [next guide](https://developer.newrelic.com/build-apps/ab-test/publish/#submit-catalog-information). I think this is a similar situation to the one @tyreer points out below.

brammerl replied:
Makes sense, I can make those changes.

brammerl commented:
@tyreer and @alexronquillo: implemented the feedback that was noted and also added some new stuff to the create-hello-world-app page. 

With regards to ab-testing: it would be a good idea to update the [publish page](https://developer.newrelic.com/build-apps/ab-test/publish/). In order to get screenshots, I would have to go through the whole process. Happy to do so but just wondering if there is anyone who already did that for the already existing screenshots. If so, they could send the file over to me and I could implement the what's inside view to get screenshots. 

This [publish your nerdpack page](https://developer.newrelic.com/build-apps/publish-deploy/publish/) also seems like a good target. I have yet to edit it. 

Let me know your thoughts.

tyreer commented on src/markdown-pages/build-apps/serve-publish-subscribe/catalog.mdx:
```diff
@@ -201,6 +201,10 @@ Click your Nerdpack to see the new details:
 
 ![App details](../../../images/serve-publish-subscribe/demo-app-details.png)
 
+Click the **What's inside** tab to see any screenshots you uploaded for individual items in your Nerdpack"
```
```suggestion
Click the **What's inside** tab to see any screenshots you uploaded for individual items in your Nerdpack
```

Little extra quote there

brammerl commented:
@tyreer finished up those last edits you suggested. Take your time in looking at them.

alexronquillo commented on src/markdown-pages/build-apps/build-hello-world-app.mdx:
```diff
@@ -246,6 +246,10 @@ nr1 catalog:submit
 
+![Overview View](../../images/create-hello-world/hello-world-overview.png)
+
+![What's Inside View](../../images/create-hello-world/hello-world-whats-inside.png)
```
@brammerl would you mind addressing my comment here, as well? I think the second screenshot here could really benefit from its own description.

jpvajda commented:
Is this still a valid PR @brammerl ?

brammerl commented:
> Is this still a valid PR @brammerl ?

@jpvajda yes it is! I think it needs to be merged in but there are snyk issues that i'm not sure how to resolve.

tyreer commented:
This is 🟢 to me! :D
